### DISCUSSION TITLE
Preserve granular topics after Hierarchical Topic Modelling

### DISCUSSION THREAD

**Discussion by noahberhe** [MaartenGr/BERTopic#1165] [Q&A]
Hello!

After running BERTopic unsupervised, I created a seed_topic_list to be passed into Guided BERTopic, then a KeyBERT representation layer.
The results are bigram topic labels which are good albeit granular.
I understand I can merge these topics, however I'd like to retain the original granular topics, in cases some people want to see that detail.

Is this possible? So that I can choose to display my docs either at the topic_high_level or the topic_low_level granularity, and also display the hierarchy?

If not possible, I was thinking of the alternatives:
Append granular bigram topics to dataframe to preserve them before running:

```
representation_model = PartOfSpeech("en_core_web_sm", pos_patterns=[[{'POS':'NOUN'}]]

topic_model.update_topics(docs,
                          n_gram_range=(1, 1),
                          top_n_words=1,
                          representation_model=representation_model) 
```

and append these as the high-level results.

But would this give a unigram that's potentially unrelated to the granular bigram, ie. if it picks out one of the representative words for that topic to construct the unigram?

Thanks
Noah

**Comment by MaartenGr** [comment]
Hierarchical options are implemented in BERTopic, like [this](https://maartengr.github.io/BERTopic/getting_started/hierarchicaltopics/hierarchicaltopics.html) which might be relevant for you. It creates a hierarchy without updating the internal model. That way, you can use the resulting dataframe of hierarchical topics to demonstrate this potential hierarchy. You would, however, need to update the model if you want to show different levels of hierarchy for every function/visualization that BERTopic supports.
